File: py/two_colors.card_game.py
n = int(input())

s_arr = []
for x in range(n):
  s_arr.append(input())

m = int(input())

t_arr = []
for x in range(m):
  t_arr.append(input())

tu = set(s_arr)

count = 0
for x in tu:
  tmp = s_arr.count(x)
  tmp_a = t_arr.count(x)
  count = max(count, tmp-tmp_a)

print(count)

File: c++/two_colors_card_game.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
  int n;
  int tmp;
  string s;
  map<string,int> mp{};

  cin >> n;

  for (int i = 0; i < n; i++) {
    cin >> s;
    if (mp.count(s)) {
      tmp = mp[s];
      tmp++;
      mp[s] = tmp;
    } else {
      mp.emplace(s,1);
    }
  }

  int m;
  cin >> m;

  for (int i = 0; i < m; i++) {
    cin >> s;
    if (mp.count(s)) {
      tmp = mp[s];
      tmp--;
      mp[s] = tmp;
    }
  }

  int mx = 0;

  for (auto i = mp.begin(); i != mp.end(); i++) {
    mx = max(i->second,mx);
  }

  cout << mx << endl;
  return 0;
}
